### PR TITLE
test(provider): fix race in test

### PR DESCRIPTION
**Pull request by hydrogen18**
I think in the process of fixing one race, I introduced another one. This refactor should fix that. I also made some changes suggested by @troian  to prevent non obvious timeouts in CI.

### CODE REVIEW

File: testutil/channel_wait.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package testutil
+
+import (
+	"reflect"
+	"testing"
+	"time"
+)
+
+func ChannelWaitForValue(t *testing.T, waitOn interface{}) interface{} {
```
Comment (troian) [inline testutil/channel_wait.go:9]: can we have here options interface allowing caller to set custom timeout value?

Comment (hydrogen18) [reply testutil/channel_wait.go:9]: Sure, I will have two functions in this file. One with a default value set and the other where the caller can set how long to wait.

Comment (troian) [reply testutil/channel_wait.go]: you can do one function with options pattern like [here](https://github.com/VolantMQ/volantmq/blob/master/connection/writerOptions.go)

Comment (hydrogen18) [reply testutil/channel_wait.go]: I don't really see an advantage to doing that for what is a single option. It's just going to create a complex way of doing an obvious thing.

**Review by boz — APPROVED** [review]

File: testutil/channel_wait.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+	delayChan := time.After(waitFor)
+
+	cases[1] = reflect.SelectCase{
+		Dir:  reflect.SelectRecv,
+		Chan: reflect.ValueOf(delayChan),
+		Send: reflect.Value{},
+	}
+
+	idx, v, ok := reflect.Select(cases)
```
Comment (Ropes) [inline testutil/channel_wait.go:25]: TIL :exploding_head:

**Review by Ropes — APPROVED** [review]

**Pull request merged**